Mathlib/Data/PFunctor/Multivariate/Basic.lean

```diff
@@ -38,122 +38,125 @@ open MvFunctor (LiftP LiftR)
 variable {n m : ℕ} (P : MvPFunctor.{u} n)
 
 /-- Applying `P` to an object of `Type` -/
+@[coe]
 def Obj (α : TypeVec.{u} n) : Type u :=
-  Σa : P.A, P.B a ⟹ α
+  Σ a : P.A, P.B a ⟹ α
 #align mvpfunctor.obj MvPFunctor.Obj
 
+instance : CoeFun (MvPFunctor.{u} n) (fun _ => TypeVec.{u} n → Type u) where
+  coe := Obj
+
 /-- Applying `P` to a morphism of `Type` -/
-def map {α β : TypeVec n} (f : α ⟹ β) : P.Obj α → P.Obj β := fun ⟨a, g⟩ => ⟨a, TypeVec.comp f g⟩
+def map {α β : TypeVec n} (f : α ⟹ β) : P α → P β := fun ⟨a, g⟩ => ⟨a, TypeVec.comp f g⟩
 #align mvpfunctor.map MvPFunctor.map
 
 instance : Inhabited (MvPFunctor n) :=
   ⟨⟨default, default⟩⟩
 
 instance Obj.inhabited {α : TypeVec n} [Inhabited P.A] [∀ i, Inhabited (α i)] :
-    Inhabited (P.Obj α) :=
+    Inhabited (P α) :=
   ⟨⟨default, fun _ _ => default⟩⟩
 #align mvpfunctor.obj.inhabited MvPFunctor.Obj.inhabited
 
-instance : MvFunctor P.Obj :=
+instance : MvFunctor.{u} P.Obj :=
   ⟨@MvPFunctor.map n P⟩
 
 theorem map_eq {α β : TypeVec n} (g : α ⟹ β) (a : P.A) (f : P.B a ⟹ α) :
     @MvFunctor.map _ P.Obj _ _ _ g ⟨a, f⟩ = ⟨a, g ⊚ f⟩ :=
   rfl
 #align mvpfunctor.map_eq MvPFunctor.map_eq
 
-theorem id_map {α : TypeVec n} : ∀ x : P.Obj α, TypeVec.id <$$> x = x
+theorem id_map {α : TypeVec n} : ∀ x : P α, TypeVec.id <$$> x = x
   | ⟨_, _⟩ => rfl
 #align mvpfunctor.id_map MvPFunctor.id_map
 
 theorem comp_map {α β γ : TypeVec n} (f : α ⟹ β) (g : β ⟹ γ) :
-    ∀ x : P.Obj α, (g ⊚ f) <$$> x = g <$$> f <$$> x
+    ∀ x : P α, (g ⊚ f) <$$> x = g <$$> f <$$> x
   | ⟨_, _⟩ => rfl
 #align mvpfunctor.comp_map MvPFunctor.comp_map
 
-instance : LawfulMvFunctor P.Obj where
+instance : LawfulMvFunctor.{u} P.Obj where
   id_map := @id_map _ P
   comp_map := @comp_map _ P
 
 /-- Constant functor where the input object does not affect the output -/
-def const (n : ℕ) (A : Type u) : MvPFunctor n :=
+def Const (n : ℕ) (A : Type u) : MvPFunctor n :=
```

Comment:
Technically, `MvPFunctor` is a structure, so the naming convention calls for this definition to be lowerCamelCase.
Then again, the point of this PR is to pretend that `MvPFunctor` is a `Type`-valued function, in which case we do UpperCamelCase.

I support this change, since it improves consistency between `MvPFunctor` and `MvFunctor`, but still I think we should discuss on a Zulip thread whether we want to treat structures with a `CoeFun` instance as if they were function types, and if so, to document this exception in the naming convention document.

Reply:
https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/UpperCamelCase.20for.60CoeFun.20.CE.B1.20.28fun.20_.20.3A.20.CE.B2.20.3D.3E.20Type*.29.60.20.237526

Reply:
From this discussion, I decided to use lowerCamelCase.